src/parser/expression.js

```diff
@@ -753,6 +769,10 @@ pp.parseObj = function (isPattern, refShorthandDefaultPos) {
     node.properties.push(prop);
   }
 
+  if (firstRestLocation !== null) {
```

Comment:
These can be moved right below the `while (!this.eat(tt.braceR)) {` block right? - I guess it was left at the end as convention? Works either way